In [6]:
!pip install -r requirements.txt

In [7]:
import numpy as np

# Sistema Masa Resorte

## Problema

Considera un sistema masa-resorte en un plano horizontal, en el cual un resorte está unido a un punto fijo 
y del otro extremo a una masa de 0.5 kg, se inicia el movimiento con el resorte estirado a 10 cm, por último, 
sabemos que la constante del resorte es de 50 N/m.
Determina el periodo del sistema.

Obtén:
* Posición
* Velocidad 
* Aceleración 

en un tiempo de un segundo.

In [45]:
m = 0.5 # masa en kg
k = 50 # constante del resorte en N/m
A = 0.1 # amplitud en m
t = 1 # tiempo en s para calcular la posición

# calculo de la frecuencia angular
omega = np.sqrt(k/m) # frecuencia angular en rad/s
print("Frecuencia angular:", f"{omega:.3f}", "rad/s")

# calculo del periodo
T = round(2 * np.pi * np.sqrt(m/k), 3) # periodo en s
print("Periodo: ", f"{T:.3f}", "s")

# calculo de la frecuencia
f = round(1/T, 3) # frecuencia en Hz
print("Frecuencia:", f, "Hz")

# calculo de la posición
x_t = round(A * np.cos(omega * t), 3)
print("Posición x(t):", x_t, "m")

# calculo de la velocidad en cm/s
v_t = round(-A * omega * np.sin(omega * t) * 100, 3)
print("Velocidad v(t):", v_t, "cm/s")

# calculo de la aceleración
a_t = round(-A * (omega ** 2) * np.cos(omega * t), 3)
print("Aceleración a(t): ", a_t, "m/s^2")


Frecuencia angular: 10.000 rad/s
Periodo:  0.628 s
Frecuencia: 1.592 Hz
Posición x(t): -0.084 m
Velocidad v(t): 54.402 cm/s
Aceleración a(t):  8.391 m/s^2


### Datos

$m = 0.5 kg$

$k = 50 N/m$

$A = 10cm = 0.1 m$

$t = 1 s$

$\phi = 0$

$\omega = \sqrt{\frac{k}{m}} = \sqrt{\frac{50}{0.5}} = 10$

# Solución

## Operaciones en base al código previo

Calculo del periodo: $T = 2\pi\sqrt{\frac{m}{k}} = 2\pi\sqrt{\frac{0.5}{50}} = 0.628 s$

Calculo de la frecuencia: $f = \frac{1}{T} = \frac{1}{0.628} = 1.592 Hz$

Calculo de la posición: $X(t) = A\cos(\omega t + \phi) = 0.1\cos(10t + \phi) = 0.1\cos(10t) = 0.1\cos(10) = 0.1(-0.839) = -0.084 m$

Calculo de la velocidad: $v(t) = -A\omega\sin(\omega t + \phi) = -0.1(10)\sin(10t + \phi) = -0.1(10)\sin(10t) = -0.1(10)\sin(10) = -0.1(0.544) = -0.05402 m/s = 54.402 cm/s$

Calculo de la aceleracion: $a(t) = -A\omega^2\cos(\omega t + \phi) = -0.1(10)^2\cos(10t + \phi) = -0.1(10)^2\cos(10t) = -0.1(10)^2\cos(10) = -0.1(10)^2(-0.839) = -8.391 m/s^2$
